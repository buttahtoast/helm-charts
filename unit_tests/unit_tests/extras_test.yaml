suite: library/extras unit tests
tests:

  - it: "[Extras | Environment]: Timezone/Proxy Extras"
    templates:
      - extras/env.yaml
    set:
      global:
        timezone: "Europe/Zurich"
        proxy:
          httpProxy:
            host: "proxy.company.com"
            port: 8080
            protocol: "http"
          httpsProxy:
            host: "proxy.company.com"
            port: 8080
            protocol: "https"
          noProxy: 
            - localhost
            - 127.0.0.1
            - company.com
    asserts:
      - matchSnapshot: {}

  - it: "[Extras | Extra Resources]: Test List"
    templates:
      - extras/list.yaml
    set:
      extraResources:
        - apiVersion: v1
          kind: Pod
          metadata:
            name: nginx
          spec:
            containers:
            - name: nginx
              image: nginx:1.14.2
              ports:
              - containerPort: 80
        - apiVersion: v1
          kind: ConfigMap
          metadata:
            name: config-multi-env-files
            namespace: default
          data:
            allowed: '"true"'
            color: purple
            enemies: aliens
            how: fairlyNice
            lives: "3"
            textmode: "true"
    asserts:
      - matchSnapshot: {}