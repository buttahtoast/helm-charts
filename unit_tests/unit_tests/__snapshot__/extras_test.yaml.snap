'[Extras | Environment]: Timezone/Proxy Extras':
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: nginx
      name: nginx-deployment
    spec:
      replicas: 3
      selector:
        matchLabels:
          app: nginx
      template:
        metadata:
          labels:
            app: nginx
        spec:
          containers:
          - env:
            - name: NODE_NAME
              valueFrom:
                fieldRef:
                  fieldPath: spec.nodeName
            - name: POD_NAME
              valueFrom:
                fieldRef:
                  fieldPath: metadata.name
            - name: POD_NAMESPACE
              valueFrom:
                fieldRef:
                  fieldPath: metadata.namespace
            - name: POD_IP
              valueFrom:
                fieldRef:
                  fieldPath: status.podIP
            - name: POD_SERVICE_ACCOUNT
              valueFrom:
                fieldRef:
                  fieldPath: spec.serviceAccountName
            - name: TZ
              value: Europe/Zurich
            - name: HTTP_PROXY
              value: http://proxy.company.com:8080
            - name: HTTPS_PROXY
              value: https://proxy.company.com:8080
            - name: NO_PROXY
              value: localhost, 127.0.0.1, company.com
            - name: JVM_OPTS
              value: -Xms512m -Xmx512m -Dhttp.proxyHost=proxy.company.com -Dhttp.proxyPort=8080
                -Dhttp.nonProxyHosts=localhost|127.0.0.1|company.com -Dhttps.proxyHost=proxy.company.com
                -Dhttps.proxyPort=8080 -Dhttps.nonProxyHosts=localhost|127.0.0.1|company.com
            image: nginx:1.14.2
            name: nginx
            ports:
            - containerPort: 80
'[Extras | Extra Resources]: Test List':
  1: |
    apiVersion: v1
    items:
    - apiVersion: v1
      kind: Pod
      metadata:
        name: nginx
      spec:
        containers:
        - image: nginx:1.14.2
          name: nginx
          ports:
          - containerPort: 80
    - apiVersion: v1
      data:
        allowed: '"true"'
        color: purple
        enemies: aliens
        how: fairlyNice
        lives: "3"
        textmode: "true"
      kind: ConfigMap
      metadata:
        name: config-multi-env-files
        namespace: default
    kind: List
